Masking st (#4261)
SumoLogic/sumologic-documentation: docs/send-data/opentelemetry-collector/remote-management/source-templates/localfile.md

---
id: localfile

> title: LocalFile Source Template

Reviewer: ```suggestion
title: Local File Source Template
```

sidebar_label: Local File
description: Learn about the Sumo Logic Local File source template for OpenTelemetry.
---

<head>
  <meta name="robots" content="noindex" />
</head>

<p><a href="/docs/beta"><span className="beta">Beta</span></a></p>

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="30"/>

> Local File source template creates an OpenTelemetry configuration that can be pushed to a remotely managed OpenTelemetry collector (abbreviated as otelcol). By creating this source template and pushing the config to the appropriate OpenTelemetry agent you can ensure collection of your logs to Sumo Logic.

Reviewer: ```suggestion
Local File source template creates an OpenTelemetry configuration that can be pushed to a remotely managed OpenTelemetry collector (abbreviated as otelcol). By creating this source template and pushing the config to the appropriate OpenTelemetry agent, you can ensure collection of your logs to Sumo Logic.
```

		
## Fields creation in Sumo Logic for Local File

If not already present, the following [Fields](/docs/manage/fields/) are created as part of source template creation.

- **`sumo.datasource`**. Fixed value of **localfile**.

> - **`deployment.environment`** User configured field at the time of collector installation. This identifies the environment where the host resides. For example: dev, prod, or qa.

Reviewer: ```suggestion
- **`deployment.environment`**. User configured field at the time of collector installation. This identifies the environment where the host resides. For example: dev, prod, or qa.
```

> - **`host.group`**. This is a collector level field and is user configured (at the time of collector installation). This identifies the host group from where logs are collected.

Reviewer: ```suggestion
- **`host.group`**. This is a collector-level field that is user configured at the time of collector installation. It identifies the host group from where logs are collected.
```

- **`host.name`**. This is tagged through the resourcedetection processor. It holds the value of the host name where the OTel collector is installed.
		
## Prerequisites
		
### For logs collection
import LogsCollectionPrereqisites from '../../../../reuse/apps/logs-collection-prereqisites.md';

<LogsCollectionPrereqisites/>

import OtelWindowsLogPrereq from '../../../../reuse/apps/opentelemetry/log-collection-prerequisite-windows.md';

<OtelWindowsLogPrereq/>

## Source template configuration
		
You can follow the below steps to set a remotely managed OpenTelemetry collector and push the source template to it.
		
### Step 1: Set up remotely managed OpenTelemetry collector
		
import CollectorInstallation from '../../../../reuse/apps/opentelemetry/collector-installation.md';
		
<CollectorInstallation/>
		
### Step 2: Configure the source template
		
In this step, you will configure the yaml required for Local File Collection. Below are the inputs required for configuration:
		
- **Name**. Name of the source template.
- **Description**. Description for the source template.

> - **Fields/Metadata**. You can provide any customer fields to be tagged with the data collected. By default sumo tags _sourceCategory with the value otel/localfile

Reviewer: ```suggestion
- **Fields/Metadata**. You can provide any customer fields to be tagged with the data collected. By default, sumo tags `_sourceCategory` with the value otel/localfile.
```

- **File Path**. Provide the file which needs to be read by OpenTelemetry agent. You can provide path to multiple files by adding new entry to it.
- **DenyList**. Provide path expression describing the files to be excluded.

> - **Collection should begin from**. Defines where will the collection of the logs start from. Possible values are - "End of File" and "Beginning of File"

Reviewer: ```suggestion
- **Collection should begin from**. Defines where will the collection of the logs start from. Possible values are "End of File" and "Beginning of File".
```

> - **File Path**. Provide the file which needs to be read by OpenTelemetry agent. You can provide path to multiple files by adding new entry to it.
> - **DenyList**. Provide path expression describing the files to be excluded.
> - **Collection should begin from**. Defines where will the collection of the logs start from. Possible values are - "End of File" and "Beginning of File"
> - **Detect messages spanning multiple lines**. You can enable this option when dealing with logs which span over multiple lines. On enabling this option you will need to specify **Boundary regex location** where you can specify if the expression defines end or start of the log line and **Expression to match message boundary** where you will define the expression. 

Reviewer: ```suggestion
- **File Path**. Provide the file path(s) that the OpenTelemetry agent should read. You can specify multiple files by adding new entries.
- **DenyList**. Provide a path expression to describe files that should be excluded.
- **Collection Start Point**. Define where the log collection should begin. Possible values are "End of File" and "Beginning of File".
- **Detect Multi-line Messages**. You can enable this option for logs that span multiple lines. When enabled, specify the **Boundary Regex Location** to indicate if the expression defines the start or end of the log line, and the **Expression to Match Message Boundary** to define the expression.
```

import OtelLogAdvanceOption from '../../../../reuse/apps/opentelemetry/logs-advance-option-otel.md';

<OtelLogAdvanceOption/>

> **Processing Rules** - **Processing Rules** -  You can add processing rules for logs collected. To learn more, refer to [Processing Rules](../processing-rules/index.md).

Reviewer: ```suggestion
**Processing Rules**. You can add processing rules for logs collected. To learn more, refer to [Processing Rules](../processing-rules/index.md).
```
